### DISCUSSION TITLE
Formulario(Submit) Practica Pasteleria

### DISCUSSION THREAD

**Discussion by UlisesOrnelasR** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#163] [Q&A] [answered]
Buen día, haciendo la práctica me surgió una duda, y es cuando nos pide "El pastelero necesita una tabla donde aparezcan los pedidos que se han realizado con los datos del formulario.".
He investigado un poco y la mayoría indican que debe usarse php y mySQL pero tenia duda porque no es parte del curso, que  harán ustedes o que me recomiendan para mostrar los resultados del formulario. Gracias!

**Comment by DanyVeneno** [comment, marked as answer]
Hola Ulises tienes razón en un proyecto real asi se hace aunque en la actualidad hay más lenguajes y herramientas que todavia no nos enseña el MCRodrigo yo sugiero que solo hagas la tabla.  saludos

**Comment by enriqueorea** [comment]
Hola, no necesariamente necesitas utilizar PHP y Mysql para el proyecto, existen muchas tecnologías que pueden realizar esa funcionalidad del proyecto, recuerda que como misión final "Back-End" nos ofrecen dos tecnologias JavaSpring y Node.JS cuya finalidad entre muchas otras, es crear funcionalidades como la que describes o la que describe el MC.

**Comment by JuanSalvador73** [comment]
De momento solo con construir el html así puro, eso del php y mySQL se ve en el BackEnd que es lo que lo vuelve dinámico. Solo ten el esqueleto para cuando mas adelante se tenga que hacer así, pero insisto, eso es mucho mas adelante. No te preocupes por eso ahora.

**Comment by avexyk** [comment]
Tienes que usar JavaScript. Para hacer la práctica un poco más dinámica busca como manipular el DOM a traves de JavaScript. En este caso por ejemplo, se me ocurre que al botón le quites el atributo submit y le agregues una función onclick. Algo así:

`<button onclick="miFuncionCaptura()">Mibotón</button>`

Una vez inicias la función optienes los datos con el id de los inputs que nos enseño el MC.

Por ejemplo:

`<input type="text" name="midato" id="midato">`

La función JavaScript quedaría así:

function miFuncionCaptura() {
  let miDatoInput = document.getElementById("midato").value;
}

La variable miDatoInput recibe el valor del input midato. Ahora hay que enviarlo al front de vuelta. Para ello podemos tener en el front un div vacío como este:

`<div id="miDivRecibeInfo"><div>`

Actualizamos la función anterior

![img13](https://user-images.githubusercontent.com/32181532/156083218-2128efab-b171-46e2-a8c8-7f4bed06287e.png)

El resultado que deberías ver es un div con varios textos dentros de etiquetas p, esta cantidad corresponde a la información que enviaste de tu formulario

Si enviaste 3 veces deberías ver algo así

`<div id="miDivRecibeInfo"><p>Texto1</p><p>Texto2</p><p>Texto3</p><div>`

Entendiendo el concepto de manipulación con JavaScript puedes notar que puedes hacerlo dinámico y crear una tabla como pide el MC. Obviamente es información que se mantendra en tu front. Y cuando des F5 volvera como al principio.

Edite varias veces, ya que por alguna razón se desaparecían las comillas al enviar el comentario. Me imagino por el Markdown

**Reply by OmarVenturaP** [reply]
Buscaba la forma y creeme que no le encontraba, gracias por compartirlo, voy a tratar de implementarlo en mi codigo. Muchas gracias por compartir

**Reply by OmarVenturaP** [reply]
no se si lo estaré haciendo bien, pero creo que no me funcionó, seguire viendo si no hice algo equivocado... Saludos

**Comment by avexyk** [comment]
Voy a subir una demo en unos momentos para que vean el funcionamiento. Entendiendo mas o menos la idea se puede crear una tabla generada de forma dinamica con los valores de un formulario sin mucho problema.

**Reply by OmarVenturaP** [reply]
Exclente, fijate que intente (no se si lo estare haciendo bien, haciendolo tanto en un archivo .js aparte o tambien con  el script dentro del mismo html) pero la verdad si en JS estoy bien perdido, la neta no tengo ninguna nocion de ello, pero agradeceria un pequeño empujon para poder implementarlo...  por si puedes hecharte una vuelta a mi repo, para que me digas como voy @avexyk, si es que puedes, claro.

**Reply by avexyk** [reply]
Ya lo termine. Algo sencillo. Solo para que se vea la idea. Se puede ver aquí en línea [demo](https://avexyk.github.io/javascript-html/) y acá esta el repositorio con el código [link](https://github.com/avexyk/javascript-html)

Edit: JavaScript no es muy complicado, solo es de entender un tanto el flujo como lo fué cuando trabajamos con Python. Hay muchas guías en internet útiles para entenderlo. Ya ví tu repo, te esta quedando genial, felicidades.

**Reply by OmarVenturaP** [reply]
Fíjate que si ya le entendí un poco a ese archivo que subiste. Voy a tratar de emularlo y te comento como va. Gracias por el dato.

Y sobre el repo ay vamos, déjame decirte que a puros videos de Youtube me la eh llevado para aprender más sobre HTML y voy apenas familiarizandome con CSS. Solo que aún no comienzo con JS, por ello se me complica más el proceso. Gracias por el comentario.
